Build fails with -DENABLE_CELX=no

Used build flags:
`cmake -G Ninja -DCMAKE_INSTALL_PREFIX=/usr -DENABLE_CELX=no -DENABLE_NLS=yes -DENABLE_GTK=OFF -DENABLE_QT5=ON -DENABLE_TOOLS=ON -DENABLE_WIN=OFF -DENABLE_FFMPEG=yes`

Compile error:
```
[86/229] /usr/bin/x86_64-pc-linux-gnu-g++ -DCONFIG_DATA_DIR=\"/usr/share/celestia\" -DEIGEN_NO_DEBUG -DENABLE_NLS -DFMT_SHARED -DGIT_COMMIT=\"67474341\" -DHIP_DATA_DIR=\"/usr/share/celestia\" -DLOCALEDIR=\"/usr/share/locale\" -DNO_DEBUG -DUSE_FFMPEG -DVERSION=\"1.7.0\" -D_USE_MATH_DEFINES -Dcelestia_EXPORTS -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999_build/src/celestia -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999_build -isystem /usr/include/eigen3 -isystem /usr/include/freetype2  -march=native -mtune=native -O2 -pipe -fsigned-char -std=gnu++17 -fPIC -fPIC -fno-exceptions -MD -MT src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o -MF src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o.d -o src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o -c /var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia/configfile.cpp
FAILED: src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o
/usr/bin/x86_64-pc-linux-gnu-g++ -DCONFIG_DATA_DIR=\"/usr/share/celestia\" -DEIGEN_NO_DEBUG -DENABLE_NLS -DFMT_SHARED -DGIT_COMMIT=\"67474341\" -DHIP_DATA_DIR=\"/usr/share/celestia\" -DLOCALEDIR=\"/usr/share/locale\" -DNO_DEBUG -DUSE_FFMPEG -DVERSION=\"1.7.0\" -D_USE_MATH_DEFINES -Dcelestia_EXPORTS -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999_build/src/celestia -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src -I/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999_build -isystem /usr/include/eigen3 -isystem /usr/include/freetype2  -march=native -mtune=native -O2 -pipe -fsigned-char -std=gnu++17 -fPIC -fPIC -fno-exceptions -MD -MT src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o -MF src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o.d -o src/celestia/CMakeFiles/celestia.dir/configfile.cpp.o -c /var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia/configfile.cpp
/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia/configfile.cpp: In function ‘CelestiaConfig* ReadCelestiaConfig(const std::filesystem::__cxx11::path&, CelestiaConfig*)’:
/var/tmp/portage/sci-astronomy/celestia-9999/work/celestia-9999/src/celestia/configfile.cpp:349:13: error: type ‘const class Value’ argument given to ‘delete’, expected pointer
  349 |      delete configParamsValue;
      |             ^~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

 - OS: Gentoo Linux
 - Version: Current git head (commit 674743410c199000791a781f07ef056b563c4b14)

--- a/src/celestia/configfile.cpp
+++ b/src/celestia/configfile.cpp
@@ -40,7 +40,7 @@ CelestiaConfig* ReadCelestiaConfig(const fs::path& filename, CelestiaConfig *con
         return config;
     }
 
-    const Value configParamsValue = parser.readValue();
+    Value configParamsValue = parser.readValue();
     const Hash* configParams = configParamsValue.getHash();
     if (configParams == nullptr)
     {
@@ -50,12 +50,6 @@ CelestiaConfig* ReadCelestiaConfig(const fs::path& filename, CelestiaConfig *con
 
     if (config == nullptr)
         config = new CelestiaConfig();
-
-#ifdef CELX
-    config->configParams = configParams;
-    if (auto path = configParams->getPath("LuaHook"); path.has_value())
-        config->luaHook = *path;
-#endif
 
     config->faintestVisible = configParams->getNumber<float>("FaintestVisibleMagnitude").value_or(6.0f);
     if (auto path = configParams->getPath("FavoritesFile"); path.has_value())
@@ -342,11 +336,11 @@ CelestiaConfig* ReadCelestiaConfig(const fs::path& filename, CelestiaConfig *con
         }
     }
 
-    // TODO: not cleaning up properly here--we're just saving the hash, not the instance of Value
-    config->params = configParams;
-
-#ifndef CELX
-     delete configParamsValue;
+#ifdef CELX
+    if (auto path = configParams->getPath("LuaHook"); path.has_value())
+        config->luaHook = *path;
+    // Move the value into the config object to retain ownership of the hash
+    config->configParams = std::move(configParamsValue);
 #endif
 
     return config;

--- a/src/celestia/configfile.h
+++ b/src/celestia/configfile.h
@@ -7,13 +7,13 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
-#ifndef _CONFIGFILE_H_
-#define _CONFIGFILE_H_
+#pragma once
 
 #include <string>
 #include <vector>
 #include <celengine/parser.h>
 #include <celengine/star.h>
+#include <celengine/value.h>
 #include <celcompat/filesystem.h>
 
 
@@ -51,7 +51,7 @@ public:
     std::string scriptSystemAccessPolicy;
 #ifdef CELX
     fs::path luaHook;
-    const Hash* configParams;
+    Value configParams;
 #endif
 
     fs::path HDCrossIndexFile;
@@ -68,8 +68,6 @@ public:
     unsigned int aaSamples;
 
     unsigned int consoleLogRows;
-
-    const Hash* params;
 
     float SolarSystemMaxDistance;
     unsigned ShadowMapSize;
@@ -89,5 +87,3 @@ public:
 };
 
 CelestiaConfig* ReadCelestiaConfig(const fs::path& filename, CelestiaConfig* config = nullptr);
-
-#endif // _CONFIGFILE_H_

--- a/src/celestia/qt/qtappwin.cpp
+++ b/src/celestia/qt/qtappwin.cpp
@@ -893,7 +893,11 @@ void CelestiaAppWindow::slotOpenScriptDialog()
     QString scriptFileName = QFileDialog::getOpenFileName(this,
                                                           _("Open Script"),
                                                           dir,
+#ifdef CELX
                                                           _("Celestia Scripts (*.celx *.cel)"),
+#else
+                                                          _("Celestia Scripts (*.cel)"),
+#endif
                                                           nullptr,
                                                           QFileDialog::DontUseNativeDialog);
 

--- a/src/celestia/win32/winmain.cpp
+++ b/src/celestia/win32/winmain.cpp
@@ -2879,7 +2879,11 @@ static void HandleOpenScript(HWND hWnd, CelestiaCore* appCore)
     ZeroMemory(&Ofn, sizeof(OPENFILENAME));
     Ofn.lStructSize = sizeof(OPENFILENAME);
     Ofn.hwndOwner = hWnd;
+#ifdef CELX
     Ofn.lpstrFilter = "Celestia Script\0*.celx;*.clx;*.cel\0";
+#else
+    Ofn.lpstrFilter = "Celestia Script\0*.cel\0";
+#endif
     Ofn.lpstrFile= szFile;
     Ofn.nMaxFile = sizeof(szFile);
     Ofn.lpstrFileTitle = szFileTitle;

--- a/src/celscript/lua/celx_celestia.cpp
+++ b/src/celscript/lua/celx_celestia.cpp
@@ -2686,8 +2686,11 @@ static int celestia_getparamstring(lua_State* l)
     CelestiaCore* appCore = this_celestia(l);
     const char* s = Celx_SafeGetString(l, 2, AllErrors, "Argument to celestia:getparamstring must be a string");
     CelestiaConfig* config = appCore->getConfig();
-    const std::string* paramString = config->configParams->getString(s);
-    lua_pushstring(l, paramString == nullptr ? "" : paramString->c_str());
+    if (auto params = config->configParams.getHash(); params != nullptr)
+    {
+        const std::string* paramString = params->getString(s);
+        lua_pushstring(l, paramString == nullptr ? "" : paramString->c_str());
+    }
     return 1;
 }
 

